src/Chainweb/CutDB.hs

```diff
@@ -431,45 +458,49 @@ startCutDb config logfun headerStore payloadStore cutHashesStore = mask_ $ do
     -- 3. exitence of dependencies
     -- 4. full validation
     --
-    readInitialCut = withTableIter (_getRocksDbCas cutHashesStore) $ \it -> do
-        tableIterLast it
-        go it
-      where
-        -- TODO: should we limit the search to a certain number of attempts
-        -- or iterate in increasinly larger steps?
-        go it = tableIterValue it >>= \case
-            Nothing -> do
-                logg Info "using initial cut from cut db configuration"
-                return $! _cutDbParamsInitialCut config
-            Just ch -> try (lookupCutHashes wbhdb ch) >>= \case
-                Left (e@(TreeDbKeyNotFound _) :: TreeDbException BlockHeaderDb) -> do
-                    logfun @T.Text Warn
-                        $ "Unable to load cut at height " <>  sshow (_cutHashesHeight ch)
-                        <> " from database."
-                        <> " Error: " <> sshow e <> "."
-                        <> " The database might be corrupted. Falling back to previous cut."
-                    tableIterPrev it
-                    go it
-                Left e -> throwM e
-                Right hm -> do
-                    limitedCut <- unsafeMkCut v
-                        <$> case _cutDbParamsInitialHeightLimit config of
-                            Nothing -> return hm
-                            Just h -> do
-                                let
-                                    cutHeightTarget = max 0 $
-                                        avgCutHeightAt v h -
-                                            CutHeight (int $ diameterAtCutHeight v (maxBound :: CutHeight) * chainCountAt v (maxBound :: BlockHeight))
```

Comment:
Can you explain why this term is removed. It's not clear what its purpose was before and why it isn't needed any more with this PR.

Reply:
The idea was... from what I remember, the idea was based on the incorrect idea that we were searching for a cut in the store, not creating the cut ourselves. If we were searching for a cut, we couldn't just ask for a cut with height equal to `blockheight * numchains`, because some block heights might be above where we wanted in the cut.

But we aren't doing that. We're fabricating the cut from the blocks. And when we ask `limitCut` to do that, we can feed it exactly `blockheight * numchains` as the number of cut heights and it will just divide out `numchains` and grab the blocks at exactly the height we want.